Gordon360/Services/ProfileService.cs

```diff
@@ -323,6 +326,47 @@ public async Task<StudentProfileViewModel> UpdateMobilePhoneNumberAsync(string u
             return profile;
         }
 
+        /// <summary>
+        /// office location setting
+        /// </summary>
+        /// <param name="username"> The username for the user whose office location is to be updated </param>
+        /// <param name="newBuilding">The new building location to update the user's office location to</param> 
+        /// <param name="newRoom">The new room to update the user's office room to</param>
+        public async Task<FacultyStaffProfileViewModel> UpdateOfficeLocationAsync(string username, string newBuilding, string newRoom)
+        {
+            var profile = GetFacultyStaffProfileByUsername(username);
+            if (profile == null)
+            {
+                throw new ResourceNotFoundException() { ExceptionMessage = "The account was not found" };
+            }
+            var user = _webSQLContext.accounts.Where(a => a.AD_Username == username).FirstOrDefault();
+            user.Building = newBuilding;
+            user.Room = newRoom;
+            await _webSQLContext.SaveChangesAsync();
+
+            return profile;
+        }
+
+        /// <summary>
+        /// office hours setting
+        /// </summary>
+        /// <param name="username"> The username for the user whose office hours is to be updated </param>
+        /// <param name="newHours">The new hours to update the user's office hours to</param>
+        public async Task<FacultyStaffProfileViewModel> UpdateOfficeHoursAsync(string username, string newHours)
+        {
+            var profile = GetFacultyStaffProfileByUsername(username);
+            if (profile == null)
+            {
+                throw new ResourceNotFoundException() { ExceptionMessage = "The account was not found" };
+            }
+            var user = _webSQLContext.accounts.Where(a => a.AD_Username == username).FirstOrDefault();
+            var user2 = _webSQLContext.account_profiles.Where(a => a.account_id == user.account_id).FirstOrDefault();
+            user2.office_hours = newHours;
+            await _webSQLContext.SaveChangesAsync();
+
+            return profile;
+        }
+
```

Comment:
```suggestion
        public async Task<FacultyStaffProfileViewModel> UpdateOfficeLocationAsync(string username, string newBuilding, string newRoom)
        {
            var profile = GetFacultyStaffProfileByUsername(username);
            if (profile == null)
            {
                throw new ResourceNotFoundException() { ExceptionMessage = "The account was not found" };
            }
            var user = _webSQLContext.accounts.Where(a => a.AD_Username == username).FirstOrDefault();
            user.Building = newBuilding;
            user.Room = newRoom;
            await _webSQLContext.SaveChangesAsync();

            return profile;
        }

        /// <summary>
        /// office hours setting
        /// </summary>
        /// <param name="username"> The username for the user whose office hours is to be updated </param>
        /// <param name="newHours">The new hours to update the user's office hours to</param>
        public async Task<FacultyStaffProfileViewModel> UpdateOfficeHoursAsync(string username, string newHours)
        {
            var profile = GetFacultyStaffProfileByUsername(username);
            if (profile == null)
            {
                throw new ResourceNotFoundException() { ExceptionMessage = "The account was not found" };
            }
            var user = _webSQLContext.accounts.Where(a => a.AD_Username == username).FirstOrDefault();
            var user2 = _webSQLContext.account_profiles.Where(a => a.account_id == user.account_id).FirstOrDefault();
            user2.office_hours = newHours;
            await _webSQLContext.SaveChangesAsync();

            return profile;
        }


a few things:
1) These routes/services can be combined if you use the `PatchViewModel` that I suggested in the other PR. 
2) `user` and `user2` are not very good variable names

```

Reply:
Thanks! I changed to a patch and changed the naming